sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/ControlFlowGraph/UcfgInstructionTest.Suspect.cs

```diff
@@ -68,6 +68,58 @@ public string Foo(Class1 c)
                 // %1 := object.ToString() [ c ]
         }
     }
+}";
+            UcfgVerifier.VerifyInstructions(code, "Foo");
+        }
+
+        [TestMethod]
+        public void NullCoalescingOperator()
+        {
+            const string code = @"
+namespace Namespace
+{
+    public class Class1
+    {
+        private Class1 field;
+         public void Foo(Class1 c)
+        {
+            var result = c ?? field;
+                // %0 := __id [ this.field ]
+                // result := __id [ const ]
+        }
+    }
+}";
+            UcfgVerifier.VerifyInstructions(code, "Foo");
+        }
+
+        [TestMethod]
+        public void NullConditionalOperator()
+        {
+            const string code = @"
+namespace Namespace
+{
+    public class BaseClass
+    {
+        public BaseClass baseField;
+    }
+     public class Class1 : BaseClass
+    {
+        private Class1 field;
+         public void Foo()
+        {
+            var result = field?.field;
+                // %0 := __id [ this.field ]
```

Comment:
This first access (read) to field comes from the previous block (linked to the null check).